File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/maps/Markers.java
package org.maplibre.android.maps;

import android.graphics.RectF;

import androidx.annotation.NonNull;

import org.maplibre.android.annotations.BaseMarkerOptions;
import org.maplibre.android.annotations.Marker;

import java.util.List;

/**
 * Interface that defines convenient methods for working with a {@link Marker}'s collection.
 */
interface Markers {
  Marker addBy(@NonNull BaseMarkerOptions markerOptions, @NonNull MapLibreMap maplibreMap);

  List<Marker> addBy(@NonNull List<? extends BaseMarkerOptions> markerOptionsList, @NonNull MapLibreMap maplibreMap);

  void update(@NonNull Marker updatedMarker, @NonNull MapLibreMap maplibreMap);

  List<Marker> obtainAll();

  @NonNull
  List<Marker> obtainAllIn(@NonNull RectF rectangle);

  void reload();
}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/maps/Markers.kt
package org.maplibre.android.maps

import android.graphics.RectF
import org.maplibre.android.annotations.BaseMarkerOptions
import org.maplibre.android.annotations.Marker

/**
 * Interface that defines convenient methods for working with a [Marker]'s collection.
 */
internal interface Markers {
    fun addBy(
        markerOptions: BaseMarkerOptions<*, *>,
        maplibreMap: MapLibreMap,
    ): Marker

    fun addBy(
        markerOptionsList: List<BaseMarkerOptions<*, *>>,
        maplibreMap: MapLibreMap,
    ): List<Marker>

    fun update(
        updatedMarker: Marker,
        maplibreMap: MapLibreMap,
    )

    fun obtainAll(): List<Marker>

    fun obtainAllIn(rectangle: RectF): List<Marker>

    fun reload()
}
